kubespawner/spawner.py

```diff
@@ -800,6 +801,93 @@ def _hub_connect_port_default(self):
         """
         )
 
+    form_template = Unicode(
+        """<label for="profile">Please select a profile for your environment:</label>
+        <select class="form-control" name="profile" required autofocus>
+        {{ inputs }}
+        </select>
+        """,
+        config = True,
+        help = """Jinja2 template to use to construct options_form text. {inputs} is replaced with
+            the result of formatting inputs against each item in the profiles list."""
+        )
+
+    inputs = Unicode("""
+        <option value="{{ key }}" {{ first }} >{{ display }}</option>""",
+        config = True,
+        help = """Jinja template to construct {inputs} in form_template. This text will be formatted
+            against each item in the profiles list, in order, using the following key names:
+            ( display, key, type ) for the first three items in the tuple, and additionally
+            first = "checked" (taken from first_input) for the first item in the list, so that
+            the first item starts selected."""
+        )
+
+    first_input = Unicode('selected',
+        config=True,
+        help="Text to substitute as {first} in inputs"
+        )
+
+    options_form = Unicode()
+
+    profile_list = List(
+        trait = Dict(),
+        default_value = None,
+        minlen = 0,
+        config = True,
+        help = """
+            List of profiles to offer for selection by the user.
+
+            Signature is: List(Dict()), where each item is a dictionary that has two keys:
+            - 'display_name': the human readable display name
+            - 'kubespawner_override': a dictionary with overrides to apply to the KubeSpawner
+              settings. Each value can be either the final value to change or a callable that
+              take the `KubeSpawner` instance as parameter and return the final value.
+
+            Example::
+
+                c.KubeSpawner.profile_list = [
+                    {
+                        'display_name': 'Training Env - Python',
+                        'kubespawner_override': {
+                            'singleuser_image_spec': 'training/python:label',
+                            'cpu_limit': 1,
+                            'mem_limit': '512M',
+                        }
+                    },
+                    {
```

Comment:
Very minor point, you've formatted the rest of the list as `}, {`

Reply:
done